packages/core-ethereum/crates/core-ethereum-api/src/executors.rs

```diff
@@ -19,6 +22,26 @@ pub trait EthereumClient<T: Into<TypedTransaction>> {
     async fn post_transaction(&self, tx: T) -> Result<Hash>;
 }
 
+/// Instantiation of `EthereumClient` using `HoprRpcOperations`.
+#[derive(Clone)]
+pub struct RpcEthereumClient<Rpc: HoprRpcOperations> {
+    rpc: Arc<Mutex<Rpc>>,
+}
+
+impl<Rpc: HoprRpcOperations> RpcEthereumClient<Rpc> {
+    pub fn new(rpc: Arc<Mutex<Rpc>>) -> Self {
```

Comment:
Cannot there be multiple instances of a non-mutexed RPC connection?

Reply:
That instance is presumed shared with the Indexer, and as it contains the contract instances, I'm not 100% convinced if it is a good idea to have those exist multiple times in memory.